packages/flutter_android_lifecycle/android/src/main/java/io/flutter/embedding/engine/plugins/lifecycle/FlutterLifecycleAdapter.java

```diff
@@ -0,0 +1,27 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.engine.plugins.lifecycle;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.Lifecycle;
+
+public class FlutterLifecycleAdapter {
+  @NonNull private final Lifecycle lifecycle;
+
+  public FlutterLifecycleAdapter(@NonNull Object reference) {
+    if (!(reference instanceof HiddenLifecycleReference)) {
+      throw new IllegalArgumentException(
+          "The reference argument must be of type HiddenLifecycleReference. Was actually "
+              + reference);
+    }
+
+    this.lifecycle = ((HiddenLifecycleReference) reference).getLifecycle();
+  }
+
+  @NonNull
+  public Lifecycle getLifecycle() {
```

Comment:
why not making this a static method?  would this class have state in the future?

Reply:
It is possible that tests will be written that would like to alter what is returned by `FlutterLifecycleAdapter`, or verify calls on it. For general testing purposes, I only make things static that absolutely need to be static, or fundamentally do not make sense as instance members.